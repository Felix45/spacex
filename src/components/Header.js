import React from 'react';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import Image from 'react-bootstrap/Image';
import LogoImage from '../assets/images/logo.png';

const Header = () => (
  <header>
    <Navbar bg="light" expand="md" fixed="top">
      <Container>
        <Navbar.Brand href="/">
          <Image height="28px" src={LogoImage} alt="Logo" />
          <span className="ms-2">SpaceX</span>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="ms-auto" activeKey="/">
            <Nav.Link href="/">Rockets</Nav.Link>
            <Nav.Link href="/missions">Missions</Nav.Link>
            <div className="vr d-none d-md-block m-2" />
            <Nav.Link href="/profile">My Profile</Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  </header>
);

export default Header;
