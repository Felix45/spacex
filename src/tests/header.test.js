import React from 'react';
import Header from '../components/Header';
import renderWithProviders from './test-utils';

describe('Header component', () => {
  it('Renders correctly', () => {
    const { asFragment } = renderWithProviders(<Header />);
    const firstRender = asFragment();

    expect(firstRender).toMatchSnapshot();
  });
});
