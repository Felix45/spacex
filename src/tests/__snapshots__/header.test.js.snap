// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders correctly 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="navbar navbar-expand-md navbar-light bg-light fixed-top"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="Logo"
            class=""
            height="28px"
            src="logo.png"
          />
          <span
            class="ms-2"
          >
            SpaceX
          </span>
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="ms-auto navbar-nav"
          >
            <div
              class="nav-link"
            >
              <a
                aria-current="page"
                class="text-primary text-decoration-none"
                href="/"
              >
                Rockets
              </a>
            </div>
            <div
              class="nav-link"
            >
              <a
                class="text-secondary text-decoration-none"
                href="/missions"
              >
                Missions
              </a>
            </div>
            <div
              class="vr d-none d-md-block m-2"
            />
            <div
              class="nav-link"
            >
              <a
                class="text-secondary text-decoration-none"
                href="/profile"
              >
                Profile
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
